### PR TITLE
Business logic to open order detail on order item click in 2 pane layout

### PR DESCRIPTION
**Pull request by AnirudhBhat**
Closes: #10588 

### Description

This PR introduces business logic to open the selected order in the detail pane

Note: This PR does not include highlighting the selected order in the order list screen

### Testing instructions

1. Run the app on physical tablet or in a tablet emulator
2. Click on the `Orders` tab
3. Select order items from the order listing screen on the left
4. Ensure the selected order's details are displayed on the detail pane on the right

### Images/gif

[pr_10592.webm](https://github.com/woocommerce/woocommerce-android/assets/1331230/5b5ed4fb-eab6-49fb-9e90-89628af3ddb1)

- [ ] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt [hunk 1]
```diff
@@ -1121,8 +1126,17 @@ class MainActivity :
             arrayOf(orderId).toLongArray(),
             remoteNoteId
         )
+        navHostFragment?.navController?.let { navController ->
+            val bundle = Bundle().apply {
+                putLong(KEY_ORDER_ID, orderId)
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt]: ❓ These keys are defined in `nav_graph_orders`. It's not safe to duplicate-hardcode the keys in code here, e.g. if they are renamed in the XML navigation will break. Can we create the `Bundle` using `OrderDetailFragmentArgs` class instead?

**Review by samiuelson — COMMENTED** [review]
Good work overall. I've added a small improvement idea. Let me know what you think.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt (hunk 1 above)
Comment (AnirudhBhat) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainActivity.kt]: Good point. Thanks for catching this! Fixed here: https://github.com/woocommerce/woocommerce-android/pull/10592/commits/52235a65a518f2274a1c5ebcdb9d15e6e64500ac

**Review by samiuelson — APPROVED** [review]
LGTM

**Pull request merged**